src/utils/workflow-utils.ts

```diff
@@ -146,60 +165,76 @@ export const generateTaskSpec = (
 
     draftSpec.endpoint.connector = connectorConfig;
 
-    if (resourceConfigs) {
+    if (!isEmpty(resourceConfigs) && !isEmpty(bindings)) {
         const collectionNameProp = getCollectionNameProp(entityType);
-
-        const boundCollectionNames = Object.keys(resourceConfigs);
-
-        boundCollectionNames.forEach((collectionName) => {
-            const resourceConfig = resourceConfigs[collectionName].data;
-
-            // Check if disable is a boolean otherwise default to false
-            const { disable } = resourceConfigs[collectionName];
-            const resourceDisable = isBoolean(disable) ? disable : false;
-
-            // See which binding we need to update
-            const existingBindingIndex = getBindingIndex(
-                draftSpec.bindings,
-                collectionName
-            );
-
-            if (existingBindingIndex > -1) {
-                // Include disable otherwise totally remove it
-                if (resourceDisable) {
-                    draftSpec.bindings[existingBindingIndex].disable =
-                        resourceDisable;
-                } else {
-                    delete draftSpec.bindings[existingBindingIndex].disable;
+        const { fullSource } = options;
+
+        Object.entries(bindings).forEach(([_collection, bindingUUIDs]) => {
+            bindingUUIDs.forEach((bindingUUID, iteratedIndex) => {
+                const resourceConfig = resourceConfigs[bindingUUID].data;
+
+                const { bindingIndex, collectionName, disable } =
+                    resourceConfigs[bindingUUID].meta;
+
+                // Check if disable is a boolean otherwise default to false
+                const bindingDisabled = isBoolean(disable) ? disable : false;
+
+                // See which binding we need to update
+                const existingBindingIndex = resourceConfigServerUpdateRequired
+                    ? getBindingIndex(
+                          draftSpec.bindings,
+                          collectionName,
+                          iteratedIndex
+                      )
```

Comment:
This is okay for now. But this approach needs to change somewhat quickly.

We should not be calling this inside of a loop. Instead we should handle fetching what collections are at what index before the loop. That way we are not running through a mapping and filtering of all the bindings every time for every binding.

Reply:
Sure. This is worth a discussion around expectations; primarily, whether `getBindingIndex` should be used in this function at all. I am inclined to say is should not be used here.

As aforementioned, I aspired to make as minimal changes as possible with the binding model refactoring. The binding index was evaluated in a similar fashion previously, so I lightly modified it in this PR.

Reply:
Yeah - we really should strive to not use it here at all. Can you throw in a `TODO` here and explain what the future work could be.